torch/fx/experimental/symbolic_shapes.py

```diff
@@ -249,6 +249,16 @@ def _sympystr(self, printer):
 
         @classmethod
         def eval(cls, base, divisor):
+            # Python supports more types (like float), but we don't do type
+            # promotion here. And it's not clear whether we need to support more
+            # types based on existing models.
+            def check_supported_type(x):
+                if not isinstance(x, FloorDiv) and not x.is_integer:
```

Comment:
This isinstance check looks very questionable. I think the real problem here is that we didn't plug in FloorDiv into sympy's assumptions system; if both inputs to floor div are integers, the output should be an integer too.

cc @Chillee